homework2/task04/project4/task4.py

```diff
@@ -0,0 +1,34 @@
+"""
+Write a function that accepts another function as an argument. Then it
+should return such a function, so the every call to initial one
+should be cached.
+
+
+def func(a, b):
+    return (a ** b) ** 2
+
+
+cache_func = cache(func)
+
+some = 100, 200
+
+val_1 = cache_func(*some)
+val_2 = cache_func(*some)
+
+assert val_1 is val_2
+
+"""
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    """Return function."""
+    cache_data = {}
+
+    def wrapper(*args: int) -> int:
```

Comment:
why int?

Reply:
Yes, you are right. It can be also string, list and so on.